### PR TITLE
Add checkstyle and overall cleaning

### PR DESCRIPTION
**Pull request by hugy718**
This PR adds a checkstyle file that should respected by all committers in the future for code formatting. 
* the code repository is cleaned to comply with this standard
* a Github action to automatically test compliance

### CODE REVIEW

**Comment by hugy718** [comment]
@KimballCai @Zrealshadow @NLGithubWP @liuchangshiye 
I have cleaned the cool-core directory. Fixed the indentation, variable names, imports, and missing java docs on public APIs. Please use the cool-checkstyle.xml file in your code editor for future development, and do a rebase to the `dev` branch, once this PR is merged.

An github action is deployed to invoke checkstyle to check the formatting on added/modified files for future PRs.

Cleaning of the other folders, namely cool-example, cool-extensions and cool-queryserver will be done later, as they do not affect ongoing developments we discussed in recent weeks.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -3,17 +3,16 @@
 
 name: Java CI with Maven
```
Comment (KimballCai) [inline .github/workflows/maven.yml:4]: maybe we can also rename this workflow.

**Comment by KimballCai** [comment]
Besides, could you share the coding style file? to define the indent and others?

File: .github/workflows/maven.yml (hunk 1 above)
Comment (hugy718) [reply .github/workflows/maven.yml:4]: sure

**Comment by hugy718** [comment]
> Besides, could you share the coding style file? to define the indent and others?

It is called cool-checkstyle.xml file under the root directory of this project. It is modified based on the Google Java coding style built-in configuration in checkstyle. I relaxed some naming rules to better suit our code base.

**Review by KimballCai — APPROVED** [review]

**Pull request merged**